File: for-local-test/367.valid-perfect-square.java

class Solution {
    public boolean isPerfectSquare(int num) {
        if (num > 0 && num < 4) return true;
        long left = 1;
        long right = num;
        while(left <= right){
            long mid = left + (right - left) / 2;
            long square = mid * mid;
            if (square == num) return true;
            else if (square < num) left = mid + 1;
            else right = mid - 1;
        }
        return false;
    }
    public static void main(String[] args) {
        Solution solution = new Solution();
        int num = 16;
        boolean result = solution.isPerfectSquare(num);
        System.out.println("Is " + num + " a perfect square? " + result);
    }
}

File: for-local-test/python/367.valid-perfect-square.py
class Solution:
    def isPerfectSquare(self, num: int) -> bool:
        if num > 0 and num < 4:
            return True
        left, right = 2, num
        while left <= right:
            mid = left + (right - left) // 2
            square = mid * mid
            if square == num:
                return True
            elif square < num:
                left = mid + 1
            else:
                right = mid - 1
        return False
    
if __name__ == "__main__":
    solution = Solution()
    num = 16
    result = solution.isPerfectSquare(num)
    print(f"Is {num} a perfect square? {result}")  # Output: True